### ISSUE TITLE
Documentation pointing to github

### ISSUE THREAD

**Issue by MichaelSchulzMETNO** [metno/pyaerocom#262]
It would be nice if the documentation is pointing directly to the file on github which can be edited. "View page source" is currently not really working. This would facilitate improving the documentation, I believe.

@jgliss

**Comment by jgliss** [comment]
Sorry in advance for a long message...

@MichaelSchulzMETNO good point, however, it requires some steps to update the online documentation at https://pyaerocom.met.no/. 



This is because the html files shown on the website are build automatically using the python auto-documentation software [sphinx](https://www.sphinx-doc.org/en/master/usage/quickstart.html), from a set of [restructuredtext files](https://docutils.sourceforge.io/rst.html) (end with .rst), which you see in "raw" when you click on "View page source" (which I believe works as it should). These .rst files are stored in the pyaerocom repo under [pyaerocom/docs](https://github.com/metno/pyaerocom/tree/master/docs), and the toplevel is index.rst where also the table of contents is defined. 

So changing something on https://pyaerocom.met.no/ requires the following steps:

1. You need to have pyaerocom and pyaerocom-docs cloned locally, in the same directory.
2. You have to have a python environment installed with sphinx library and extensions to be able to run sphinx.
3. Change the corresponding .rst file in pyaerocom/docs
4. From within pyaerocom/docs, recompile the documentation using sphinx: 

   ```bash
   $ make clean
   $ make html
   ```

   This will create all html files displayed on the website and stores them in the [pyaerocom-docs](https://github.com/metno/pyaerocom-docs) repo which needs to be cloned locally in the same directory where pyaerocom repo is cloned).

5. Push updated pyaerocom-docs, which is then pulled on the webserver once per day automatically (the latter is setup by @jgriesfeller)

**NOTE**: the documentation website depends also on the pyaerocom branch, for instance, I updated the .rst file pyaerocom/docs/readme.rst today from one of the development branches, thus, what will be displayed at pyaerocom.met.no from tonight is not corresponding to the master branch...

Currently, this is all a little cumbersome and cannot really be done automatically, as we have these manual steps of using pyaerocom-docs repo and pushing and pulling it on the webserver. If we would use an integrated service such as ReadTheDocs (I believe NorESM uses this service), only step 3. would be needed and everything else would be done automatically, whenever someone commits a change in the pyaerocom repository.

Perhaps we should change this setup and use ReadTheDocs. @jgriesfeller  mentioned that there might be licensing issues with using services such as readthedocs  as an institution, and that we might need to pay for that. So we should check the terms of readthedocs again. It could be that it is free for open source software packages, and it would make life easier to keep the documentation synchronised with the pyaerocom repo. Also because it will handle different realisations of the docs for different branches individually.

**One more general comment: please never change something in the master branch directly but make a new branch for every change you do and create a pull request into the destination branch (and assign someone else to review this change). Note that usually the master branch corresponds to the latest release, so when you update something (e.g. in the documentation), you probably want to branch off the `main-dev` branch and create the pull-request into `main-dev` accordingly. Whenever a new release is done, `main-dev` is merged into `master`. This way we ensure that the current documentation website corresponds to the master branch and the latest official release.**

**Comment by MichaelSchulzMETNO** [comment]
Thanks - three small remarks:

Would it be good to have a "how to contribute ?" section in the documentation with the remarks above ?

Would it be good to have a contribute.md file in the master code directory?

And what I meant for edit github links is something like what I found here:
https://gist.github.com/mgedmin/6052926

**Comment by MichaelSchulzMETNO** [comment]
As discussed, it would be probably useful to explore if switching to readthedocs could allow for
- more easy automatic documentation updates
- more interactive link between github code and documentation for contritbutors
- allow for documentation of different releases

**Comment by jgliss** [comment]
@MichaelSchulzMETNO I agree with you on all points, it will be good to put a contribute.md file that explains all details. Before that we should find out if ReadTheDocs is an option. 

It would be great if others could easily help with improving the documentation website.
